packages/dialog/index.tsx

```diff
@@ -0,0 +1,322 @@
+      eventTargetMatches: (target: HTMLElement, selector: string) => matches(target, selector),
+      trapFocus: () => this.focusTrap && this.focusTrap.activate(),
+      releaseFocus: () => this.focusTrap && this.focusTrap.deactivate(),
+      isContentScrollable: () => {
+        const content = this.content_;
+        return (!!content) ? isScrollable(content) : false;
+      },
+      areButtonsStacked: () => areTopsMisaligned(this.buttons_),
+      getActionFromEvent: (evt: any) => {
+        const elem = closest(evt.target, `[${strings.ACTION_ATTRIBUTE}]`);
+        return elem && elem.getAttribute(strings.ACTION_ATTRIBUTE);
+      },
+      clickDefaultButton: () => {
+        const defaultButton = this.defaultButton_;
+        if (defaultButton) {
+          defaultButton.click();
+        }
+      },
+      reverseButtons: () => {
+        const buttons = this.buttons_;
+        return buttons &&
+          buttons.reverse().forEach((button: HTMLButtonElement) =>
+            button.parentElement && button.parentElement.appendChild(button)
+          );
+      },
+      notifyOpening: () => this.handleOpening(),
+      notifyOpened: () => this.props.onOpen && this.props.onOpen(),
+      notifyClosing: (action: string) => this.handleClosing(action),
+      notifyClosed: (action: string) => this.props.onClose && this.props.onClose(action),
+    };
+  }
+
+  handleOpening = (): void => {
+    this.props.onOpening!();
+    LAYOUT_EVENTS.forEach((evt: string) =>
+      window.addEventListener(evt, this.handleLayout)
+    );
+    document.addEventListener(
+      'keydown', this.handleDocumentKeyDown
+    );
+  }
+
+  handleClosing = (action: string): void => {
+    this.props.onClosing!(action);
+    LAYOUT_EVENTS.forEach((evt: string) =>
+      window.removeEventListener(evt, this.handleLayout)
+    );
+    document.removeEventListener(
+      'keydown', this.handleDocumentKeyDown
+    );
+  }
+
+  handleInteraction = (e: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>): void =>
+    this.foundation.handleInteraction(e);
+  handleDocumentKeyDown = (e: Event): void =>
+    this.foundation.handleDocumentKeydown(e);
+  handleLayout = (): void => this.foundation.layout();
+
+
+  render() {
+    const {
+      /* eslint-disable no-unused-vars */
+      autoStackButtons,
+      className,
+      children,
+      escapeKeyAction,
+      id,
+      onOpening,
+      onOpen,
+      onClosing,
+      onClose,
+      open,
+      scrimClickAction,
+      tag: Tag,
+      ...otherProps
+      /* eslint-enable no-unused-vars */
+    } = this.props;
+
+    const container: React.ReactElement<HTMLDivElement> | undefined =
+      this.renderContainer(children);
+    return (
+      // @ts-ignore
+      <Tag
+        {...otherProps}
+        aria-labelledby={this.labelledBy}
+        aria-describedby={this.describedBy}
+        aria-modal
+        className={this.classes}
+        id={id}
+        role='alertdialog'
```

Comment:
I think this should be overrideable....if you set this on otherProps, will it override role?

Reply:
don't forget this one too